src/renderer/base/renderer.cpp

```diff
@@ -217,7 +217,7 @@ void Renderer::TriggerSystemRedraw(const til::rect* const prcDirtyClient)
 // - <none>
 void Renderer::TriggerRedraw(const Viewport& region)
 {
-    auto view = _viewport;

```

Comment:
HAH. But then, shouldn't we get rid of the entire cached viewport? We're just kicking the "torn state" can down the road...

Reply:
In the past scrolling was entirely implicit in the renderer as it computed the scroll and its invalidation just from the previous and current viewport. Nowadays we also got explicit scrolling, presumably for ConPTY. Unfortunately (and similar to other parts here), the old code never got removed and the new code only got added. That's why we need to retain the behavior for both.